packages/core-components/api-report.md

```diff
@@ -1511,6 +1511,12 @@ export const useSidebarOpenState: () => SidebarOpenState;
 // @public
 export const useSidebarPinState: () => SidebarPinState;
 
+// @public
+export const useSignInProviderElements: (
+  providers: IdentityProviders,
+  onSignInSuccess: (userIdentity: UserIdentity) => void,
+) => (boolean | React_2.JSX.Element[])[];
```

Comment:
I think that we probably want to change the returned API a little bit here from the `useSignInProviders`.

It looks like it returns an array of `loading` and `elements`.

https://github.com/backstage/backstage/blob/8743beff4d843cc0fe0896a04407369f2db5c902/packages/core-components/src/layout/SignInPage/providers.tsx#L203

I think that we should probably make that API a little bit friendlier. Maybe

`return { loading, elements }` or something instead?